app/services/identity_linker.rb

```diff
@@ -1,10 +1,13 @@
 IdentityLinker = Struct.new(:user, :provider) do
   attr_reader :identity
 
-  def link_identity(nonce: nil, session_uuid: nil, ial: nil)
+  def link_identity(session_uuid: nil, **extra_attrs)
     find_or_create_identity
 
-    identity.update!(identity_attributes(nonce: nonce, session_uuid: session_uuid, ial: ial))
+    attributes = identity_attributes(session_uuid: session_uuid).
```

Comment:
thoughts on throwing this logic into an `attributes` method (that requires an arg) so the merging of attributes logic can be separated from `link_identity` ?